tmail-backend/webadmin/webadmin-email-address-contact/src/test/java/com/linagora/tmail/webadmin/EmailAddressContactRoutesTest.java

```diff
@@ -524,4 +524,141 @@ void deleteContactShouldOnlyUseMailAddressLocalPart() {
+            .then()
+                .statusCode(NO_CONTENT_204);
+
+            assertThat(Flux.from(emailAddressContactSearchEngine.list(CONTACT_DOMAIN))
+                    .map(EmailAddressContact::fields)
+                    .collectList()
+                    .block())
+                .containsOnly(new ContactFields(new MailAddress(mailAddressA), "Bob", "Loop"));
+        }
+
+        @Test
+        void updateContactShouldCreateContactWhenNotExists() throws Exception {
+            String request = "{" +
+                "  \"firstname\": \"John\"," +
+                "  \"surname\": \"Carpenter\"" +
+                "}";
+
+            given()
+                .body(request)
+            .when()
+                .put("/john")
+            .then()
+                .statusCode(NO_CONTENT_204);
+
+            assertThat(Flux.from(emailAddressContactSearchEngine.list(CONTACT_DOMAIN))
+                    .map(EmailAddressContact::fields)
+                    .collectList()
+                    .block())
+                .containsOnly(new ContactFields(new MailAddress(mailAddressA), firstnameA, surnameA));
+        }
+
+        @Test
+        void updateContactShouldOnlyUseMailAddressLocalPart() {
+            String request = "{" +
+                "  \"firstname\": \"John\"," +
+                "  \"surname\": \"Carpenter\"" +
+                "}";
+
+            Map<String, Object> errors = given()
+                .body(request)
+            .when()
+                .put("/" + mailAddressA)
+            .then()
+                .statusCode(BAD_REQUEST_400)
+                .extract()
+                .body()
+                .jsonPath()
+                .getMap(".");
+
+            assertThat(errors)
+                .containsEntry("statusCode", BAD_REQUEST_400)
+                .containsEntry("type", "InvalidArgument")
+                .containsEntry("message", "Mail address is wrong. Be sure to include only the local part in the path");
+        }
+
+        @Test
+        void updateContactShouldSucceedWhenSurnameMissing() throws Exception {
+            ContactFields contactFields = new ContactFields(new MailAddress(mailAddressA), firstnameA, surnameA);
+            Mono.from(emailAddressContactSearchEngine.index(CONTACT_DOMAIN, contactFields)).block();
+
+            String request = "{" +
+                "  \"firstname\": \"Bobby\"" +
+                "}";
+
+            given()
+                .body(request)
+            .when()
+                .put("/" + contactFields.address().getLocalPart())
+            .then()
+                .statusCode(NO_CONTENT_204);
+
+            assertThat(Flux.from(emailAddressContactSearchEngine.list(CONTACT_DOMAIN))
+                    .map(EmailAddressContact::fields)
+                    .collectList()
+                    .block())
+                .containsOnly(new ContactFields(new MailAddress(mailAddressA), "Bobby", ""));
```

Comment:
Can we keep the surname/firstname when the according input field is null?

I am afraid that when the is an UPDATE OBJECT on LSC not related to contact (for other events like update mailAlias,  we can miss the firstname/surname field) => We would reset the names while we do not intend to.
E.g:
```
mainid: user@james.org
operation: UPDATE_OBJECT
sources: REPLACE_VALUES
 - alias-to-user@james.org
 - alias-to-user_bis@james.org
```
Another E.g:
```
mainid: user@james.org
operation: UPDATE_OBJECT
firstname: REPLACE_VALUES
 - Bobby
```
Here we just want to update firstname => With this current API, we would be unintended to reset surname also.

Not sure though, what do you guys think? LSC expert @vttranlina ?

Reply:
It is an edge case, I don't think in real life we have a case remove firstname, surname to empty. 
So we can ignore updating the  firstname/surname when null

Reply:
That would quite complexify the code TMail side honestly... It means if some fields are missing I would need to go fetch the contact details first, and then fill up missing first or surname with what's in the contact...

So... LSC does a kind of patch then for updates? (partial updates)

Reply:
> LSC does a kind of patch then for updates?

It's a rare case. But it is possible I think. So the question here is should we support that partial update? If not we may encounter missing data when updating partial from the LSC side.

Reply:
Ok I will try to take a look...

Reply:
Let me know if that answers your concern :)

Reply:
I think it does. I will check against LSC. Thank you :')